cmd/migrate/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log/slog"
+	"os"
+
+	"gorm.io/gorm"
+
+	"github.com/getAlby/hub/db"
+)
+
+func main() {
+	var fromDSN, toDSN string
+
+	flag.StringVar(&fromDSN, "from", "", "source DSN")
+	flag.StringVar(&toDSN, "to", "", "destination DSN")
+
+	flag.Parse()
+
+	if fromDSN == "" || toDSN == "" {
+		flag.Usage()
+		slog.Error("missing DSN")
+		os.Exit(1)
+	}
+
+	stopDB := func(d *gorm.DB) {
+		if err := db.Stop(d); err != nil {
+			slog.Error("failed to close database", "error", err)
+		}
+	}
+
+	slog.Info("opening source DB...")
+	fromDB, err := db.NewDB(fromDSN, false)
+	if err != nil {
+		slog.Error("failed to open source database", "error", err)
+		os.Exit(1)
+	}
+	defer stopDB(fromDB)
+
+	slog.Info("opening destination DB...")
+	toDB, err := db.NewDB(toDSN, false)
+	if err != nil {
+		slog.Error("failed to open destination database", "error", err)
+		os.Exit(1)
+	}
+	defer stopDB(toDB)
+
+	slog.Info("migrating...")
+	err = migrateDB(fromDB, toDB)
+	if err != nil {
+		slog.Error("failed to migrate database", "error", err)
+		os.Exit(1)
+	}
+
+	slog.Info("migration complete")
+}
+
+func migrateDB(from, to *gorm.DB) error {
+	tx := to.Begin()
+	defer tx.Rollback()
+
+	if err := tx.Error; err != nil {
+		return fmt.Errorf("failed to start transaction: %w", err)
+	}
+
+	// Table migration order matters: referenced tables must be migrated
```

Comment:
can we fail the migration if the number of tables in the DB are not expected (e.g. we made a new table and forgot to update this migrate command?)